### PR TITLE
Add URLRedirect model to conman.redirects

### PR DESCRIPTION
**Pull request by meshy**
~~Needs to be rebased once #88 is merged~~ done.

~~Might be worth holding off on this one until #86 has been addressed.~~

- [x] Rebase once #86 is merged.

### CODE REVIEW

File: conman/redirects/models.py [hunk 1]
```diff
@@ -26,3 +26,10 @@ def save(self, *args, **kwargs):
         """Validate the Redirect before saving."""
         self.clean()
         return super().save(*args, **kwargs)
+
+
+class URLRedirect(Route):
+    """A `Route` that redirects to an arbitrary URL."""
+    handler = handlers.URLRedirectHandler.path()
+    target = models.URLField(max_length=2000)
+    permanent = models.BooleanField(default=False, blank=True)
```
Comment (Ian-Foote) [inline conman/redirects/models.py]: The `blank=True` isn't strictly necessary - [Django forces it in `BooleanField.__init__`](https://github.com/django/django/blob/1.10.4/django/db/models/fields/__init__.py#L977).

File: conman/redirects/tests/test_views.py [hunk 2]
```diff
@@ -49,3 +49,33 @@ def test_access_redirect(self):
...
+
+
+class TestURLRedirectView(RequestTestCase):
+    """Verify behaviour of URLRedirectView."""
+    view = views.URLRedirectView
+
+    def test_target(self):
+        """URLRedirect redirects to the target url."""
+        target = 'https://example.com/'
+        route = URLRedirectFactory.create(target=target)
```
Comment (Ian-Foote) [inline conman/redirects/tests/test_views.py]: Can we use `.build`?

File: conman/redirects/views.py [hunk 3]
```diff
@@ -14,3 +14,17 @@ def get_redirect_url(self, *args, **kwargs):
         redirect = kwargs['route']
         self.permanent = redirect.permanent
         return redirect.target.url
+
+
+class URLRedirectView(RedirectView):
```
Comment (Ian-Foote) [inline conman/redirects/views.py]: I think a docstring could be useful.

File: conman/redirects/views.py [hunk 4]
```diff
@@ -14,3 +14,17 @@ def get_redirect_url(self, *args, **kwargs):
...
+    permanent = False  # Set to django 1.9's default to avoid RemovedInDjango19Warning
+
+    def get_redirect_url(self, *args, **kwargs):
+        """
+        Return the target url.
+
+        Save the route's redirect type for use by RedirectView.
+        """
+        redirect = kwargs['route']
```
Comment (Ian-Foote) [inline conman/redirects/views.py]: Should route be an explicit keyword argument?

File: conman/redirects/models.py (hunk 1 above)
Comment (meshy) [inline conman/redirects/models.py]: Ooh, that's good to know, thanks!

**Comment by meshy** [comment]
Updated

**Pull request merged**